From 81257eed2316971197e2b4053cf9ac51734834df Mon Sep 17 00:00:00 2001
From: Toma Tabacu <toma.tabacu@imgtec.com>
Date: Mon, 8 Dec 2014 14:55:09 +0000
Subject: [PATCH] MIPS: LLVMLinux: Fix a 'cast to type not present in union'
 error.

Replace a 16-bit unsigned int member bitfield with an unsigned short member
variable in the 'mips16e_instruction' union.

This follows the example set by the 'mips_instruction' union, which doesn't use
bitfields for its generic union members.

This error gets reported by clang, but not by gcc.
The changed code can be compiled successfully by both gcc and clang.

Signed-off-by: Toma Tabacu <toma.tabacu@imgtec.com>
---
 arch/mips/include/uapi/asm/inst.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/mips/include/uapi/asm/inst.h b/arch/mips/include/uapi/asm/inst.h
index 4bfdb9d..3d692c0 100644
--- a/arch/mips/include/uapi/asm/inst.h
+++ b/arch/mips/include/uapi/asm/inst.h
@@ -910,7 +910,7 @@ union mips_instruction {
 };
 
 union mips16e_instruction {
-	unsigned int full : 16;
+	unsigned short full;
 	struct m16e_rr rr;
 	struct m16e_jal jal;
 	struct m16e_i64 i64;
-- 
1.9.1

